tools/validate-nebula-package.js

```diff
@@ -0,0 +1,115 @@
+const fs = require('fs');
+const path = require('path');
+
+const validateScripts = (pkg) => {
+  if (pkg.scripts.build !== 'node ./tools/build.js --core && shx cp assets/* dist') {
+    throw new Error('package.json does not have correct build script');
+  }
+  if (pkg.scripts.prepublishOnly !== 'NODE_ENV=production yarn run build && yarn spec') {
+    throw new Error('package.json does not have correct prepublishOnly script');
+  }
+  if (pkg.scripts.prepack !== './tools/prepare-sn-pack.js') {
+    throw new Error('package.json does not have correct prepack script');
+  }
+};
+
+const validatePackageJsonContent = (pkg) => {
+  // == VALIDATE package.json contents ==
+  // package name
+  console.log(pkg.name);
+  if (!pkg.name.match(/^@nebula\.js\/sn-/)) {
+    throw new Error("Bad package name. Package name should match '@nebula.js/sn-'");
+  }
+
+  if (!pkg.scripts) {
+    throw new Error('package.json does not have any build script');
+  }
+
+  if ((/--ext/.test(pkg.scripts.build) && !pkg.qext) || (!/--ext/.test(pkg.scripts.build) && pkg.qext)) {
+    throw new Error('package.json has a mismatch of qext and --ext');
+  }
+
+  if (/--native/.test(pkg.scripts.build)) {
+    throw new Error(
+      "package.json build script is not allowed to use '--native' for a chart that is not a known native chart in Qlik Sense"
+    );
+  }
+
+  validateScripts(pkg);
+};
+
+const validateFiles = (pkg) => {
+  const whitelist = [
+    'name',
+    'version',
+    'description',
+    'author',
+    'license',
+    'keywords',
+    'publishConfig',
+    'repository',
+    'files',
+    'engines',
+    'main',
+    'scripts',
+    'devDependencies',
+    'peerDependencies',
+  ];
+  // files
+  const mustHaveFiles = ['dist', 'core', 'api-specifications'];
+  if (pkg.qext) {
+    const qextName = /^@nebula\.js\/([a-z-]+)$/.exec(pkg.name)[1];
+    mustHaveFiles.push('qext');
+    mustHaveFiles.push(`${qextName}.qext`);
+    mustHaveFiles.push(`${qextName}.js`);
+  }
+  const allowedFiles = ['assets', ...mustHaveFiles];
+  const missing = mustHaveFiles.filter((f) => (pkg.files || []).indexOf(f) === -1);
+  if (missing.length) {
+    throw new Error(`package.json is missing files: ${missing.join(', ')}`);
+  }
+  const violates = (pkg.files || []).filter((f) => allowedFiles.indexOf(f) === -1);
+  if (violates.length) {
+    throw new Error(`package.json must not contain files: ${violates.join(', ')}`);
+  }
+
+  Object.keys(pkg).forEach((key) => {
+    if (whitelist.indexOf(key) === -1) {
+      delete pkg[key];
+    }
+  });
+};
+
+// known charts that are native (build time) in Sense
+const validate = (pkg, dir) => {
+  validatePackageJsonContent(pkg);
+  validateFiles(pkg);
+
+  const cleanedPkg = JSON.stringify(pkg, null, 2);
+  // package version must be 0.x.x at the moment
+  if (!/1\.\d\.\d+/.test(pkg.version)) {
```

Comment:
Care to elaborate on the purpose here? So you are not allowed to have any other version than 1.x.x? Even if the error msg and comment says something else. So version 2.x.x should not be allowed?

Reply:
Not sure, it is copy pasted from table. i assume it is blocking doing `major` version bumps. And then when it was actually bumped the regex was changed. I'll look closer at it

Reply:
Ideally, IMO, version should be based on conventional commits. In which case there are other toolsets to deal with making sure the correct version is set.

Reply:
Other than that, I'm not familiar with these reasoning behind these scripts but I see that they have already been approved for the table, so I see no reason to not allow them here as well.

Reply:
Yes, I know. I think that was Miralems suggestion back when I did the KPI. Didn't have enough knowledge/time to do it then, kind of the same now. I think we should come up with a good way to release the charts that are in seperate repos, and then use that for all charts for consistency

Reply:
I'm kind of looking at that now actually, tho it will be a simple processes initially.

Reply:
the comment is wrong.  should be // package version must be 1.x.x at the moment

Reply:
I just changed it to a more general check, so that it is only digits divided with dots. Just to catch if it is on the wrong format